src/valid/expression.rs

```diff
@@ -412,26 +430,22 @@ impl super::Validator {
                 }
 
                 // check constant offset
-                if let Some(const_handle) = offset {
-                    let good = match module.constants[const_handle].inner {
-                        crate::ConstantInner::Scalar {
-                            width: _,
-                            value: crate::ScalarValue::Sint(_),
-                        } => num_components == 1,
-                        crate::ConstantInner::Scalar { .. } => false,
-                        crate::ConstantInner::Composite { ty, .. } => {
-                            match module.types[ty].inner {
-                                Ti::Vector {
-                                    size,
-                                    kind: Sk::Sint,
-                                    ..
-                                } => size as u32 == num_components,
-                                _ => false,
-                            }
-                        }
+                if let Some(const_expr) = offset {
+                    let good = match module.const_expressions[const_expr] {
```

Comment:
This would have required that `ValidationFlags::EXPRESSIONS` depended on `ValidationFlags::CONSTANTS`.

But with the new changes (populating `const_expression_types` regardless of validation flags) this works now.